src/JWT/IJsonSerializer.cs

```diff
@@ -21,6 +21,14 @@ public interface IJsonSerializer
         /// <param name="json">The JSON string deserialize.</param>
         /// <returns>Strongly-typed object.</returns>
         object Deserialize(Type type, string json);
+
+        /// <summary>
+        /// Merges two objects into a single JSON string.
+        /// </summary>
+        /// <param name="obj1">The first object to serialize</param>
+        /// <param name="obj2">The second object to serialize, note that any existing property in obj1 will be
+        /// <returns>JSON string</returns>
+        string MergeObjects(object obj1, object obj2);

```

Comment:
I feel it's too hacky, in other words - leaky abstraction. Is there a way to support a complex model without adding this method?

Reply:
Fair enough, the solution before this commit also works so if that's a more acceptable solution we can revert back to that.

Reply:
Yes, please. Let ship a beta version and it works for everyone - then let's keep it simple.

However i cannot stress enough my appreciation for your contributions and trying to make the library better over and over again! hope you're ok with my ask to revert.

Reply:
Alright, no worries. It was an interesting solution but perhaps not the smallest code change 😀

I'll try get this sorted when I get the time, hopefully this evening.

Reply:
I have reverted that commit and rebased against latest on main now